### PR TITLE
Delay chart creation until containing element is visible.

### PR DESCRIPTION
**Pull request by anutron**
The diff here is a little hard to parse because the indentation for everything changed. The nutshell is that I've added a custom initializer that checks if the element is visible and, if not, delays creating the chart. It does this by periodically re-checking the size of the element and only invoking the initializer when it's not 0x0.

@davywentworth

### CODE REVIEW

**Comment by davywentworth** [comment]
?w=1 is your friend when whitespace is an issue ;)

can those changes be DRYed up a bit? having identical code in two places is hard to maintain. 
is it okay if that loop runs forever if that element never shows up?
other than that, :sheep:

**Comment by anutron** [comment]
I thought about abstracting it, but really the right place to put it is into Behavior itself as a startup option for filters. So I figured we'd see if we can come up with a use case that would justify that. Pushing it into Behavior would just take a lot more time. I've been pretty fastidious about writing docs, tests, and demos of any functionality added to it, so it'd take a good portion of a day to do it...

Re: element never shows up, it'll run until the element is removed from the DOM or the page is unloaded. It's a very, very inexpensive check. It's not actually doing any sort of DOM search or anything, and it's not computing anything, just literally querying a property of the element.
